ood-portal-generator/templates/ood-portal.conf.erb

```diff
@@ -85,7 +85,7 @@ Listen <%= addr_port %>
   CustomLog "<%= @accesslog %>" combined
   <%- end -%>
 
-  <%- if @servername && @use_rewrites -%>
+  <%- if !@multiple_hosts && @servername && @use_rewrites -%>
   RewriteEngine On
   RewriteCond %{HTTP_HOST} !^(<%= @proxy_server %>(:<%= @port %>)?)?$ [NC]
   RewriteRule ^(.*) <%= @ssl ? "https" : "http" %>://<%= @proxy_server %>:<%= @port %>$1 [R=301,NE,L]
```

Comment:
Why not add more `RewriteCond`s for all server aliases?

Something that ends up being like this:
```
RewriteCond %{HTTP_HOST} ^olddomain.com [NC]
RewriteCond %{HTTP_HOST} ^www.olddomain.com [NC]
RewriteCond %{HTTP_HOST} ^newdomain.com [NC]
```

Reply:
Though I do wonder how you get into this spot or why we have this redirect at all.

@treydock do we need this redirect?  If you have multiple vhosts in apache - you wouldn't hit this block at all right? (you'd hit the other directives in the other vhost). 

I guess I'm wondering under what conditions you'd actually want this to happen. If you're trying to use a different `%{HTTP_HOST}` why would apache be routing you to this vhost?

Reply:
I am not an Apache expert, but local after testing with the `ood-portal.conf` Apache config with a single `VirtualHost` directive, all domains that point to the server IP are served by the same VirtualHost, ignoring the `ServerName` and `ServerAlias`.

I guess as it is the only `VirtualHost` config it becomes the default.

Reply:
If you prefer, I can update to use the @server_aliases as you mentioned in the previous comment. Initially I thought that adding the aliases might change how it currently works, but because of the redirect taking precedence, I don't think it does.

Reply:
> I guess as it is the only VirtualHost config it becomes the default.

This is correct.  A single VirtualHost will serve all traffic pointed at Apache.

> @treydock do we need this redirect? If you have multiple vhosts in apache - you wouldn't hit this block at all right? (you'd hit the other directives in the other vhost).

The purpose with these rewrites was so that if you have this single VirtualHost, and go to like `ood-foo.example.com` but the ServerName is `ood.example.com` the rewrite would ensure all web addresses end up being the ServerName.

You'd still hit this ReWrite if ood-portal is the first VirtualHost loaded and the traffic doesn't match any of the other VirtualHosts.  In Apache the order of configs is really important so if you had like `z-example.conf` that had `ServerName foo.example.com` and the `ood-portal.conf` was `ServerName ood.example.com` then any address pointed at this server that didn't match `foo.example.com` or `ood.example.com` would end up being rewritten as `ood.example.com` and served by OnDemand.

Reply:
OK so we do need it, or should keep it.

Wouldn't something like this have the same behavior? (I'm just trying to avoid more and more and more configs)
```erb
<%- (@server_aliases + [@proxy_server]).each do |server| -%>
RewriteCond %{HTTP_HOST} !^(<%= server %>(:<%= @port %>)?)?$ [NC]
<%- end -%>
```

Reply:
This solution works for our use case. I have updated the PR with your suggestion.